pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1110,30 +1110,52 @@ func (n *NetworkProvider) reconcileRDSVpcConfiguration(ctx context.Context, priv
 		return errorUtil.Wrap(err, "error building subnet group name")
 	}
 
-	foundSubnet, err := getRDSSubnetGroup(n.RdsApi, subnetGroupName)
-	if err != nil {
-		return errorUtil.Wrap(err, "failed getting rds subnet group")
-	}
-	if foundSubnet != nil {
-		logger.Infof("subnet group %s found", *foundSubnet.DBSubnetGroupName)
-		// todo ensure subnets and tags are correct
-		return nil
+	// build array list of all vpc private subnets
+	var subnetIDs []*string
+	for _, subnet := range privateVPCSubnets {
+		subnetIDs = append(subnetIDs, subnet.SubnetId)
 	}
 
 	// in the case of no private subnets being found, we return a less verbose error message compared to obscure aws error message
 	if len(privateVPCSubnets) == 0 {
 		return errorUtil.New("no private subnets found, can not create subnet group for rds")
 	}
 
-	// build array list of all vpc private subnets
-	var subnetIDs []*string
-	for _, subnet := range privateVPCSubnets {
-		subnetIDs = append(subnetIDs, subnet.SubnetId)
+	croOwnerTag, err := getCloudResourceOperatorOwnerTag(ctx, n.Client)
+	if err != nil {
+		return errorUtil.Wrap(err, "failed to build cro owner tag")
 	}
 
-	clusterID, err := resources.GetClusterID(ctx, n.Client)
+	foundSubnet, err := getRDSSubnetGroup(n.RdsApi, subnetGroupName)
 	if err != nil {
-		return errorUtil.Wrap(err, "error getting clusterID")
+		return errorUtil.Wrap(err, "failed getting rds subnet group")
+	}
+	if foundSubnet != nil {
+		logger.Infof("subnet group %s found, verifying it is in the expected state", *foundSubnet.DBSubnetGroupName)
+		if foundSubnet.DBSubnetGroupDescription != aws.String(defaultSubnetGroupDesc) {
+			_, err := n.RdsApi.ModifyDBSubnetGroup(&rds.ModifyDBSubnetGroupInput{
+				DBSubnetGroupDescription: aws.String(defaultSubnetGroupDesc),
+				DBSubnetGroupName:        aws.String(*foundSubnet.DBSubnetGroupName),
```

Comment:
This will equate to

```suggestion
				DBSubnetGroupName:        foundSubnet.DBSubnetGroupName,
```

as we're dereferencing and then using `aws#String(string) *string` to reference to a pointer again.

All `aws#String(string) *string` is doing is:

```
// String returns a pointer to the string value passed in.
func String(v string) *string {
	return &v
}
```